lightning/src/chain/keysinterface.rs

```diff
@@ -231,18 +231,15 @@ pub trait ChannelKeys : Send+Clone {
 	/// Note that if signing fails or is rejected, the channel will be force-closed.
 	//
 	// TODO: Document the things someone using this interface should enforce before signing.
-	// TODO: Add more input vars to enable better checking (preferably removing commitment_tx and
-	// making the callee generate it via some util function we expose)!
-	fn sign_counterparty_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, feerate_per_kw: u32, commitment_tx: &Transaction, keys: &PreCalculatedTxCreationKeys, htlcs: &[&HTLCOutputInCommitment], secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()>;
+	fn sign_counterparty_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &CommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()>;
 
 	/// Create a signature for a holder's commitment transaction. This will only ever be called with
 	/// the same holder_commitment_tx (or a copy thereof), though there are currently no guarantees
 	/// that it will not be called multiple times.
 	/// An external signer implementation should check that the commitment has not been revoked.
 	//
 	// TODO: Document the things someone using this interface should enforce before signing.
-	// TODO: Add more input vars to enable better checking (preferably removing commitment_tx and
-	fn sign_holder_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, holder_commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
+	fn sign_holder_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()>;
```

Comment:
I'm a little confused about the new signatures here - they're not documented (but code indicates its the same as the sign_holder_htlc stuff), and, more importantly, don't appear to be used except in testing? Is this teeing up some future changes or did I miss something?

Reply:
Right, there is no reason not to do all of the signatures at once - it saves a round trip.  I plan to switch to fully use the new signature (and remove the other call) in a followup PR.

Reply:
Hmm, it may be a relatively large change to do this on the `OnChainTxHandler`-end, given its realtively unused, how much work is it to drop it from this patch until then?

Reply:
It's not hard, I'll drop it for now.

Reply:
done